mkosi

```diff
@@ -457,6 +457,34 @@ def install_arch(args, workspace, run_build_script):
 
     subprocess.run(cmdline, check=True)
 
+    subprocess.run(["arch-chroot", os.path.join(workspace, "root"), "systemctl", "enable", "systemd-networkd"], check=True)
+    subprocess.run(["arch-chroot", os.path.join(workspace, "root"), "systemctl", "enable", "systemd-resolved"], check=True)
+    os.remove(os.path.join(workspace, "root", "etc/resolv.conf"))
+    os.symlink("../usr/lib/systemd/resolv.conf", os.path.join(workspace, "root", "etc/resolv.conf"))
+
+    nsswitch_old = os.path.join(workspace, "root", "etc/nsswitch.conf")
+    nsswitch_new = os.path.join(workspace, "root", "etc/nsswitch.conf.new")
+
+    with open(nsswitch_old, "r") as old:
+        with open(nsswitch_new, "w") as new:
+            for line in old:
+                if line.startswith("hosts:"):
+                    new.write(" ".join(["resolve" if x == "dns" else x for x in line.split(" ")]))
+                else:
+                    new.write(line)
+
+    shutil.copystat(nsswitch_old, nsswitch_new)
+    os.remove(nsswitch_old)
+    shutil.move(nsswitch_new, nsswitch_old)
```

Comment:
i really don't like patching of config files like this I must say, but I have no better idea either... But could you please move this into a separate function, so that we can call it it also for the other distros that need this (specifically: fedora). Same for the resolv.conf handling.